airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/config/SyncBeanFactory.kt

```diff
@@ -179,4 +188,32 @@ class SyncBeanFactory {
     @Singleton
     @Named("defaultDestinationTaskLauncherHasThrown")
     fun defaultDestinationTaskLauncherHasThrown(): AtomicBoolean = AtomicBoolean(false)
+
+    /* *****
+     * CHECK
+     * *****/
+
+    // TODO combine this with the actual catalog factory method? where even is that method
```

Comment:
@johnny-schmidt do you know where the normal catalog comes from? it feels super spooky to just have this thing sitting in SyncBeanFactory

Reply:
found it here - https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/command/DestinationCatalog.kt#L53-L62

... maybe we should at least move this new bean into that file, and switch to a conditional thing instead of doing a new bean + secondary bean?